frontendVfsUtils/src/main/java/com/virtuslab/gitmachete/frontend/file/GitVfsUtils.java

```diff
@@ -0,0 +1,44 @@
+package com.virtuslab.gitmachete.frontend.file;
+
+import java.nio.file.Path;
+
+import com.intellij.openapi.vfs.VirtualFile;
+import git4idea.GitUtil;
+import git4idea.repo.GitRepository;
+import io.vavr.control.Option;
+
+public final class GitVfsUtils {
+
+  private static final String MACHETE_FILE_NAME = "machete";
+
+  private GitVfsUtils() {}
+
+  public static VirtualFile getMainDirectory(GitRepository gitRepository) {
```

Comment:
These methods look like good candidates for https://projectlombok.org/features/experimental/ExtensionMethod

Reply:
Buuuut... there's no IntelliJ support for that feature yet, ignore the comment then :/ See https://github.com/mplushnikov/lombok-intellij-plugin/issues/21